microprofile/config-smallrye/src/main/java/org/wildfly/extension/microprofile/config/smallrye/ConfigSourceDefinition.java

```diff
@@ -0,0 +1,168 @@
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.MODULE;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.NAME;
+import static org.wildfly.extension.microprofile.config.smallrye._private.MicroProfileConfigLogger.ROOT_LOGGER;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Map;
+
+import io.smallrye.config.PropertiesConfigSource;
+import org.eclipse.microprofile.config.spi.ConfigSource;
+import org.jboss.as.controller.AbstractAddStepHandler;
+import org.jboss.as.controller.AbstractRemoveStepHandler;
+import org.jboss.as.controller.AttributeDefinition;
+import org.jboss.as.controller.AttributeMarshaller;
+import org.jboss.as.controller.AttributeMarshallers;
+import org.jboss.as.controller.AttributeParsers;
+import org.jboss.as.controller.ObjectTypeAttributeDefinition;
+import org.jboss.as.controller.OperationContext;
+import org.jboss.as.controller.OperationFailedException;
+import org.jboss.as.controller.PersistentResourceDefinition;
+import org.jboss.as.controller.PropertiesAttributeDefinition;
+import org.jboss.as.controller.SimpleAttributeDefinitionBuilder;
+import org.jboss.as.controller.descriptions.ModelDescriptionConstants;
+import org.jboss.dmr.ModelNode;
+import org.jboss.dmr.ModelType;
+import org.jboss.modules.Module;
+import org.jboss.modules.ModuleIdentifier;
+import org.wildfly.extension.microprofile.config.smallrye._private.MicroProfileConfigLogger;
+
+/**
+ * @author <a href="http://jmesnil.net/">Jeff Mesnil</a> (c) 2017 Red Hat inc.
+ */
+class ConfigSourceDefinition extends PersistentResourceDefinition {
+
+    static AttributeDefinition ORDINAL = SimpleAttributeDefinitionBuilder.create("ordinal", ModelType.INT)
+            .setDefaultValue(new ModelNode(100))
+            .setAllowNull(true)
+            .setRestartAllServices()
+            .build();
+    static AttributeDefinition PROPERTIES = new PropertiesAttributeDefinition.Builder("properties", true)
+            .setAttributeParser(new AttributeParsers.PropertiesParser(false))
+            .setAttributeMarshaller(new AttributeMarshallers.PropertiesAttributeMarshaller(null, false))
+            .setAlternatives("class", "dir")
+            .setAllowNull(true)
+            .setRestartAllServices()
+            .build();
+    static ObjectTypeAttributeDefinition CLASS = ObjectTypeAttributeDefinition.Builder.of("class",
+            create(NAME, ModelType.STRING, false)
+                    .setAllowExpression(false)
+                    .build(),
+            create(MODULE, ModelType.STRING, false)
+                    .setAllowExpression(false)
+                    .build())
+            .setAlternatives("properties", "dir")
+            .setAllowNull(true)
+            .setAttributeMarshaller(AttributeMarshaller.ATTRIBUTE_OBJECT)
+            .setRestartAllServices()
+            .setCapabilityReference("org.wildfly.management.path-manager")
```

Comment:
Should this reference be here or on the `DIR` attribute?

Reply:
fixed